src/panels/Menu/TopMenuBar/Menus/ViewMenu.tsx

```diff
@@ -62,15 +86,34 @@ export function ViewMenu() {
       </TopBarMenuWrapper>
 
       <Menu.Item leftSection={<SaveIcon />}>Load/Save Layout</Menu.Item>
-      <Menu.Divider />
-      <Menu.Label>
-        <Group>
-          <VisibilityIcon /> User Visibility
-        </Group>
-      </Menu.Label>
-      <Container>
-        <Property uri={'OpenSpaceEngine.PropertyVisibility'} />
-      </Container>
+      <TopBarMenuWrapper
+        targetTitle={
+          <Menu.Item
+            leftSection={<VisibilityIcon />}
+            rightSection={<ChevronRightIcon size={IconSize.sm} />}
+          >
+            User Visibility
+          </Menu.Item>
+        }
+        position={'right-start'}
+        withinPortal={false}
+        closeOnItemClick={false}
+      >
+        <Menu.Label>Set visibility level for user</Menu.Label>
+        {options.map((option, i) => (
+          <Menu.Item
+            key={option}
+            leftSection={
+              <Group>
+                <CheckboxIndicator checked={i === propertyVisibility} />
+              </Group>
+            }
+            onClick={() => onChange(option)}
+          >
```

Comment:
Should this be Radio Buttons instead? I was thinking if there is any difference for a screen reader, otherwise I think it looks good so no biggie

Reply:
looks good